tests/test_catalog.py

```diff
@@ -256,25 +172,121 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
                         "collection": {
                             "type": "ARCHIVE",
                             "title": "title",
-                            "name": collection,
-                            "host": {"name": host},
+                            "name": self.collection,
+                            "host": {"name": self.host},
                         },
                         "productConfiguration": {"title": "config"},
                     }
                 ]
             },
         )
 
-        missing_quicklook_url = f"{constants.API_HOST}/catalog/{host}/image/{missing_image_id}/quicklook"
+    @pytest.mark.usefixtures("data_products")
+    def test_should_search(self, requests_mock: req_mock.Mocker):
+        search_url = f"{constants.API_HOST}/catalog/hosts/oneatlas/stac/search"
+        next_page_url = f"{search_url}/next"
+        bbox = (1.0, 2.0, 1.0, 2.0)
+        feature = {
+            "type": "Feature",
+            "properties": {
+                "some": "data",
+            },
+            "geometry": {"type": "Point", "coordinates": (1.0, 2.0)},
+        }
+        first_page = {
+            "type": "FeatureCollection",
+            "features": [feature],
+            "links": [
+                {
+                    "rel": "next",
+                    "href": next_page_url,
+                }
+            ],
+        }
+        second_page = {**first_page, "links": []}
+        requests_mock.post(url=search_url, json=first_page)
+        requests_mock.post(url=next_page_url, json=second_page)
+
+        results = self.catalog.search(mock_search_parameters)
+
+        assert isinstance(results, gpd.GeoDataFrame)
+        assert results.__geo_interface__ == {
+            "type": "FeatureCollection",
+            "features": [
+                {**feature, "id": "0", "bbox": bbox},
+                {**feature, "id": "1", "bbox": bbox},
+            ],
+            "bbox": bbox,
+        }
+
+    def test_search_usagetype(self):
```

Comment:
This test is to be cleaned up, together with the rest coverage for constructs_search_parameters